appendices/F/3_fix_string/palindrome/palindrome.go

```diff
@@ -0,0 +1,28 @@
+package palindrome
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// IsPalindromeNumber returns if an integer is readable in both ways.
+// Only unsigned int are supported.
+// "1221" is a palindrome
+// "01" is not a palindrome
+// "-1" is not a palindrome
+func IsPalindromeNumber(s string) (bool, error) {
+	// check if it's a number, e.g.: "ң\xa3" is not a number
+	_, err := strconv.Atoi(s)
+	if err != nil {
+		// returns an error if it's not a number
+		return false, fmt.Errorf("conversion failed %s: %w", s, err)
+	}
+
+	for i := 0; i < len(s)/2; i++ {
```

Comment:
I think this should iterate over `[]rune(s)`, not over `s`. The impact is minimal for our Pali function, but it's slightly better with `[]rune`:


```go
runes := []rune(s)
length := len(runes)
for idx := 0; idx < length / 2; idx++ {
  if runes[idx] != runes[length - 1 - idx] {
    return false, nil
  }
}
```